miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+        return self.data["brush_life_level2"]
+
+    @property
+    def brush_left_time3(self) -> int:
+        return self.data["brush_left_time3"]
+
+    @property
+    def brush_life_level3(self) -> int:
+        return self.data["brush_life_level3"]
+
+    @property
+    def device_fault(self) -> FaultStatus:
+        try:
+            return FaultStatus(self.data["device_fault"])
+        except ValueError:
+            _LOGGER.error("Unknown FaultStatus (%s)", self.data["device_fault"])
+            return FaultStatus.Unknown
+
+    @property
+    def charging_state(self) -> ChargingState:
+        try:
+            return ChargingState(self.data["charging_state"])
+        except ValueError:
+            _LOGGER.error("Unknown ChargingStats (%s)", self.data["charging_state"])
+            return ChargingState.Unknown
+
+    @property
+    def sweep_mode(self) -> SweepMode:
+        try:
+            return SweepMode(self.data["sweep_mode"])
+        except ValueError:
+            _LOGGER.error("Unknown SweepMode (%s)", self.data["sweep_mode"])
+            return SweepMode.Unknown
+
+    @property
+    def cleaning_mode(self) -> CleaningMode:
+        try:
+            return CleaningMode(self.data["cleaning_mode"])
+        except ValueError:
+            _LOGGER.error("Unknown CleaningMode (%s)", self.data["cleaning_mode"])
+            return CleaningMode.Unknown
+
+    @property
+    def sweep_type(self) -> SweepType:
+        try:
+            return SweepType(self.data["sweep_type"])
+        except ValueError:
+            _LOGGER.error("Unknown SweepType (%s)", self.data["sweep_type"])
+            return SweepType.Unknown
+
+    @property
+    def path_mode(self) -> PathMode:
+        try:
+            return PathMode(self.data["path_mode"])
+        except ValueError:
+            _LOGGER.error("Unknown PathMode (%s)", self.data["path_mode"])
+            return PathMode.Unknown
+
+    @property
+    def mop_present(self) -> bool:
+        return self.data["mop_present"]
+
+    @property
+    def work_station_freq(self) -> int:
+        return self.data["work_station_freq"]
+
+    @property
+    def timing(self) -> str:
+        return self.data["timing"]
+
+    @property
+    def clean_area(self) -> int:
+        return self.data["clean_area"]
+
+    @property
+    def uid(self) -> int:
+        return self.data["uid"]
+
+    @property
+    def auto_boost(self) -> int:
+        return self.data["auto_boost"]
+
+    def parseForbidMode(self, val):
+        def secToClock(val):
+            hour = math.floor(val / 3600)
+            minut = math.floor((val - hour * 3600) / 60)
+            return "{}:{:02}".format(hour, minut)
+
+        asDict = json.loads(val)
+        active = bool(asDict["time"][2])
+        begin = secToClock(asDict["time"][0])
+        end = secToClock(asDict["time"][1])
+        return json.dumps(
+            {"enabled": active, "begin": begin, "end": end, "tz": asDict["tz"]}
+        )
```

Comment:
Instead of using json dumps, would it be possible to use the `DNDStatus` from `vacuumcontainers.py` for this? 

If not, please create a separate container class (derived from `DeviceStatus`).